share/functions/type.fish

```diff
@@ -1,77 +1,58 @@
-
-function type --description "Print the type of a command"
+function type --description 'Print the type of a command'
+    # For legacy reasons, no argument simply causes an unsuccessful return.
+    if not set -q argv[1]
+        return 1
+    end
 
     # Initialize
     set -l res 1
     set -l mode normal
     set -l multi no
     set -l selection all
-    set -l IFS \n\ \t
 
     # Parse options
     set -l names
-    if test (count $argv) -gt 0
-        for i in (seq (count $argv))
-            set -l arg $argv[$i]
-            set -l needbreak 0
-            while test -n $arg
-                set -l flag $arg
-                set arg ''
-                switch $flag
-                    case '--*'
-                        # do nothing; this just prevents it matching the next case
-                    case '-??*'
-                        # combined flags
-                        set -l IFS
-                        echo -n $flag | read __ flag arg
-                        set flag -$flag
-                        set arg -$arg
+    while set -q argv[1]
```

Comment:
No - I had used that pattern earlier for `abbr`. It's really the nicest way to support the "--" separator.